packages/client/hmi-client/src/components/documents/tera-document-asset.vue

```diff
@@ -267,6 +269,9 @@ const toggleOptionsMenu = (event) => {
 watch(
 	() => props.assetId,
 	async () => {
+		// unsub and sub to avoid multiple subscriptions when switching between document assets
+		await unsubscribe(ClientEventType.Extraction, subscribeToExtraction);
```

Comment:
I think you just need to subscribe on mount and unsubscribe on unmount of the component. And then you can filter the event by asset Id in the event handlers. I'm not sure if it's necessary to sub and unsub on every assetId change.

Reply:
good idea - ill take a look into implementing this

Reply:
I don't know if it's worth doing more right now as this will be using the notifications system in the short term

Reply:
It looks like we are handling finished event separately here which is still needed since notification system should not  care what should happen inside this component.

Reply:
ive made changes so that we subscrive onmounted and unsub when the component is unmounted - as pointed out its probably better than doing it on asset id.

Reply:
🤔 Once your notification manager is done @jryu01 we could use the composable to keep track of certain notification events, isn't it?

Reply:
The notification manger handles the events globally only for the notification stuff. If another component wants to do  some other operations based on the events from server, it needs to subscribe for the events and handle it for its own purpose. In that way we can keep components decoupled. In this case, we want to refresh the component on complete extraction event. So I think it makes sense to register its own event handler that does that.